controller/lsp_core.ml

```diff
@@ -265,7 +267,26 @@ let do_position_request ~postpone ~params ~handler =
     (Request.Data.PosRequest { uri; handler; point; version; postpone })
 
 let do_hover = do_position_request ~postpone:false ~handler:Rq_hover.hover
-let do_goals = do_position_request ~postpone:true ~handler:Rq_goals.goals
+
+(* We get the format from the params *)
+let get_pp_format_from_config () =
+  match !Fleche.Config.v.pp_type with
+  | 1 -> Rq_goals.Pp
+  | _ -> Rq_goals.Str
```

Comment:
Do we want to do this or fail on pp types we don't know. I thinking people might use this to see if pp is supported, but it would give a false positive which might cause headaches.

Reply:
At the very least we should output a warning in the log.

Reply:
Yes, it is a pity using deriving_yojson on such unitary types doesn't work as we'd like, hopefully we can improve this part of the serialization.